core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -598,6 +598,88 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("run with ShuffleMapStage retry") {
```

Comment:
+1 on a more specific test name! I don't see any advantages of a shorter name -- just makes it hard to understand what the test does.